recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MsixConan(ConanFile):
+    name = "msix"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/msix-packaging"
+    description = "An SDK for creating MSIX packages"
+    topics = ("msix", "sdk", "packaging", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_lib": ["crypt32", "openssl"],
+        "pack": [True, False],
+        "skip_bundles": [True, False],
+        "use_external_zlib": [True, False],
+        "use_validation_parser": [True, False],
+        "xml_parser": ["applexml", "javaxml", "msxml6", "xerces"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "crypto_lib": "openssl",
+        "pack": False,
+        "skip_bundles": False,
+        "use_external_zlib": True,
+        "use_validation_parser": False,
+        "xml_parser": "xerces"
+    }
+
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.os == "Android":
+            self._cmake.definitions["AOSP"] = True
+        if self.settings.os == "Linux":
+            self._cmake.definitions["LINUX"] = True
+        if self.settings.os == "Macos":
+            self._cmake.definitions["MACOS"] = True
+        self._cmake.definitions["CRYPTO_LIB"] = self.options.crypto_lib
+        self._cmake.definitions["MSIX_PACK"] = self.options.pack
+        self._cmake.definitions["MSIX_SAMPLES"] = False
+        self._cmake.definitions["MSIX_TESTS"] = False
+        self._cmake.definitions["SKIP_BUNDLES"] = self.options.skip_bundles
+        self._cmake.definitions["USE_EXTERNAL_ZLIB"] = self.options.use_external_zlib
+        self._cmake.definitions["USE_SHARED_ZLIB"] = self.options["zlib"].shared
+        self._cmake.definitions["USE_VALIDATION_PARSER"] = self.options.use_validation_parser
+        self._cmake.definitions["XML_PARSER"] = self.options.xml_parser
+        self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
I don't know what the default crypt_lib on Windows is, but if it is bcrypt32, then you can change it like this.
```suggestion
            del self.options.fPIC
        if self.settings.os == "Windows":
            self.options.crypto_lib = "crypt32"
```

Reply:
The authors of `msix` allow to build it for Windows not only with `crypt32` but also with `openssl`, and I'm not sure it's good to restrict this option.

Reply:
it doesn't restrict this option, it changes default value on Windows, consumers can still change this option.

Reply:
Thanks, it was quite surprising that defaults can be set not only within `default_options`. [Done](https://github.com/conan-io/conan-center-index/pull/6475/commits/b082828bdc6aa1d1d99a75f30c25910c75f5e46f)